Sources/Player.swift

```diff
@@ -441,7 +458,7 @@ extension Player {
         let keys: [String] = [PlayerTracksKey, PlayerPlayableKey, PlayerDurationKey]
 
         self._asset.loadValuesAsynchronously(forKeys: keys, completionHandler: { () -> Void in
-            DispatchQueue.main.sync(execute: { () -> Void in
+            DispatchQueue.main.async(execute: { () -> Void in
```

Comment:
It should be `async` since you don't need to block the thread. I would also add a getter to the asset, so you can retrieve the asset after setting it using a url. https://github.com/efremidze/Player/tree/hotfix/asset